sqlmesh/utils/metaprogramming.py

```diff
@@ -267,6 +267,7 @@ def build_env(
     obj: t.Any,
     *,
     env: t.Dict[str, t.Any],
+    env_metadata: t.Set[str],
```

Comment:
Do we need a new attribute for this? Can we make the `env`'s value a tuple maybe?

Reply:
Thanks for spotting this, refactored and it looks cleaner now :)